codalab/client/json_api_client.py

```diff
@@ -0,0 +1,369 @@
+import httplib
+import socket
+import sys
+import urllib2
+
+from codalab.client.rest_client import RestClient, RestClientException
+from codalab.common import http_error_to_exception, UsageError
+
+
+def wrap_exception(message):
+    def decorator(f):
+        def wrapper(*args, **kwargs):
+            try:
+                return f(*args, **kwargs)
+            except urllib2.HTTPError as e:
+                # Translate known errors to the standard CodaLab errors
+                exc = http_error_to_exception(e.code, e.read())
+                if isinstance(exc, UsageError):
```

Comment:
Why just this type? What about permissions errors? Why do we even need to translate?